### DISCUSSION TITLE
Where is the documentation of the Emitter plugin

### DISCUSSION THREAD

**Discussion by lalugue** [matteobruni/tsparticles#4727] [Q&A] [answered]
Hi,

The link to the Emitter plugin seems to be missing; it shows a 404: https://particles.js.org/docs/classes/Plugins_Emitters_Options_Classes_Emitter.Emitter.html

I noticed this while going through the documentation of the `Confetti` preset: https://github.com/matteobruni/tsparticles/blob/main/presets/confetti/README.md

> The confetti cannons are made using the Emitter plugin, which is already documented [here](https://particles.js.org/docs/classes/Plugins_Emitters_Options_Classes_Emitter.Emitter.html).

Where might its new link be?

PS: I was about to make a new issue for it until I saw https://github.com/matteobruni/tsparticles/issues/4725 just a few minutes ago 😅, would it help if I still create an issue for it regardless?

I also wouldn't mind fixing the confetti readme's link with the new link

**Comment by matteobruni** [comment, marked as answer]
> Hi,
> 
> The link to the Emitter plugin seems to be missing; it shows a 404: https://particles.js.org/docs/classes/Plugins_Emitters_Options_Classes_Emitter.Emitter.html
> 
> I noticed this while going through the documentation of the `Confetti` preset: https://github.com/matteobruni/tsparticles/blob/main/presets/confetti/README.md
> 
> > The confetti cannons are made using the Emitter plugin, which is already documented [here](https://particles.js.org/docs/classes/Plugins_Emitters_Options_Classes_Emitter.Emitter.html).
> 
> Where might its new link be?
> 
> PS: I was about to make a new issue for it until I saw #4725 just a few minutes ago 😅, would it help if I still create an issue for it regardless?
> 
> I also wouldn't mind fixing the confetti readme's link with the new link

Hi @lalugue, unfortunately there's no documentation available yet. TypeDoc has an issue when creating documentation with large monorepos like this. I asked the TypeDoc owner how to achieve this and he told me that I needed around 25GB of RAM for generating the whole repository documentation, but even with 30GB I couldn't, the HeapOverflow exception appeared anyway (https://github.com/TypeStrong/typedoc/issues/1606).

Until TypeDoc is fixed, I don't know how to include that documentation again on the website. These are the methods added to the `Container` class by the Emitters plugin: https://github.com/matteobruni/tsparticles/blob/main/plugins/emitters/src/EmitterContainer.ts

Some demo configurations file are these:

- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitter.json
- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitterAbsorber.json
- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitterAngled.json
- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitterImages.json
- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitterPaths.json
- https://github.com/matteobruni/tsparticles/blob/main/demo/vanilla/public/presets/emitterShapes.json

Hope that all these resources are helpful

**Comment by lalugue** [comment]
Hi @matteobruni , thank you for the heads up and for pointing towards the presets; was originally looking for descriptions/explanations for the emitter properties via the documentation to understand it better, but at least the shared presets and the methods will help as a point of reference in the meantime while playing around with emitters

Hopefully the people in TypeDoc find a solution for larger docs too

On a side note, this also looks like an opportunity for me to play around with TypeDoc locally (and perhaps a subset of the docs)

In this case, the preset configs will help moving forward in the meantime

Thank you again 🙏

**Comment by lalugue** [comment]
by the way, it looks like Wayback Machine got a [snapshot](https://web.archive.org/web/20210903211756/https://particles.js.org/docs/classes/Plugins_Emitters_Options_Classes_Emitter.Emitter.html) of the page from last year, which seems to get the property descriptions from [Emitters.md](https://github.com/matteobruni/tsparticles/blob/main/markdown/Options/Plugins/Emitters.md)

if the descriptions in the file are still accurate, might we be able to directly link this markdown file in [confetti/README.md](https://github.com/matteobruni/tsparticles/blob/main/presets/confetti/README.md) in the meantime as a workaround/fix to the 404?

**Reply by matteobruni** [reply]
The older versions are incomplete, but maybe a dedicated webpage can be added to the website so it’s updatable until the documentation will be generated
